rfcs/text/0049-entity-fields.md

```diff
@@ -0,0 +1,158 @@
+# 0000: Entity Field Set
+
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)**
+- Date: **TBD**
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+
+<!--
+Stage 1: If the changes include field additions or modifications, please create a folder titled as the RFC number under rfcs/text/. This will be where proposed schema changes as standalone YAML files or extended example mappings and larger source documents will go as the RFC is iterated upon.
+-->
+
+<!--
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+An entity represents a discrete, identifiable component within an IT environment that can be described by a set of attributes and maintains its identity over time. Entities can be physical (like hosts or devices), logical (like containers or processes), or abstract (like applications or services).
+
+Currently, ECS provides specific field sets for certain categories of entities (e.g., host, user, cloud, orchestrator) to capture their metadata. However, as IT infrastructure continues to evolve, we encounter an increasing number of entity types that don't cleanly fit into existing field sets – for example, storage services like S3, database instances like DynamoDB, or various other cloud services and IT related infrastructure components (both digital and physical).
+
+This field set aims to solve several key challenges:
+1. Providing a flexible way to represent different types of entities without requiring new field sets for each category
+2. Supporting a consistent structure for capturing entity metadata across different entity types
+3. Enabling the representation of entities that don't fit into existing field sets
+4. Reducing the complexity of the ECS schema by avoiding the proliferation of entity-specific field sets
+
+This approach would allow ECS to accommodate new types of entities without requiring continuous schema expansion through new field sets, while maintaining a consistent structure for entity representation.
+
+## Fields
+
+| Field | Type | Description |
+|-------|------|-------------|
+| entity.id | keyword | A unique identifier for the entity. This should be a stable, unique value that persists across different observations of the same entity. For entities with dedicated field sets (e.g., host.id, user.id), this value should match the corresponding *.id field. |
```

Comment:
do we want to solve the problem of different id/name sources for entities? eg. EC2 instance id vs arn?

Reply:
I've updated the entity.id field description to provide guidance for choosing between multiple identifiers. To be transparent - there will always be some ambiguity in this selection that we can't fully resolve yet as it depends on specific use cases and contexts. However, we can provide these basic criteria for selecting the primary identifier:

1. Persists across the entity's lifecycle
2. Ensures uniqueness within its scope
3. Is commonly used for queries and correlation
4. Is readily available in most observations (events/logs, etc.,)

Alternative identifiers are preserved in entity.raw. Wdyt?

Reply:
The issue with ids is something beyond the scope of this RFC, in my opinion. More important than knowing what type of id is being persisted, is having a consistent id of that entity when ingested from different sources. I don't think adding a `id type` field would ensure consistency, it only gives a context in which I'm not sure it's needed.

We need consistent ids and entity resolution when needed. Neither of these solutions touch the RFC right now, as far as I understand.